src/layers/ArrayBitmapLayer.js

```diff
@@ -0,0 +1,117 @@
+import { BitmapLayer } from '@deck.gl/layers';
+import { Texture2D, Model, Geometry } from '@luma.gl/core';
+import GL from '@luma.gl/constants';
+
+import { PHOTOMETRIC_INTERPRETATIONS } from '../constants';
+
+const DEFAULT_TEXTURE_PARAMETERS = {
+  [GL.TEXTURE_MIN_FILTER]: GL.LINEAR,
+  [GL.TEXTURE_MAG_FILTER]: GL.LINEAR,
+  [GL.TEXTURE_WRAP_S]: GL.CLAMP_TO_EDGE,
+  [GL.TEXTURE_WRAP_T]: GL.CLAMP_TO_EDGE
+};
+
+/**
+ * This layer renders RGB interleaved data.
+ * @param {Object} props
+ * @param {Function} props.onHover Hook function from deck.gl to handle hover objects.
+ * @param {Function} props.onClick Hook function from deck.gl to handle clicked-on objects.
+ * @param {Object} props.loader A loader with the property `photometricInterpretation`.
+ * @param {Object} props.channelData { height, width, data } object where data is an array of typed arrays containing image data.
+ *
+ */
+
+export default class ArrayBitmapLayer extends BitmapLayer {
+  _getModel(gl) {
+    const {
+      loader: { photometricInterpretation }
+    } = this.props;
+    let photometricInterpretationShader;
+    // This is a port to the GPU of a subset of https://github.com/geotiffjs/geotiff.js/blob/master/src/rgb.js
+    // Safari was too slow doing this off of the GPU and it is noticably faster on other browsers as well.
+    switch (photometricInterpretation) {
+      case PHOTOMETRIC_INTERPRETATIONS.RGB:
```

Comment:
can we set a default prop for the `photometricInterpretation === PHOTOMETRIC_INTERPRETATIONS.RGB` and remove loader as a prop? 

In this case our bitmap layer would default to the original behavior of `BaseBitmapLayer`:

```javascript
import { BitmapLayer, PHOTOMETRIC_INTERPRETATIONS } from '@hms-dbmi/viv';

const layer = new BitmapLayer({ image }); // default RGB
const layer = new BitmapLayer({ image, photometricInterpretation: PHOTOMETRIC_INTERPRETATIONS.BlackIsZero });
```

Reply:
This way it will only throw if someone explicitly passes an invalid interpretation.